scene/main/window.cpp

```diff
@@ -2209,18 +2168,34 @@ void Window::start_resize(DisplayServer::WindowResizeEdge p_edge) {
 
 Rect2i Window::get_usable_parent_rect() const {
 	ERR_READ_THREAD_GUARD_V(Rect2i());
-	ERR_FAIL_COND_V(!is_inside_tree(), Rect2());
-	Rect2i parent_rect;
+
+	// When embedded we cannot resize the window.
 	if (is_embedded()) {
-		parent_rect = get_embedder()->get_visible_rect();
-	} else {
-		const Window *w = is_visible() ? this : get_parent_visible_window();
-		//find a parent that can contain us
-		ERR_FAIL_NULL_V(w, Rect2());
+		return get_embedder()->get_visible_rect();
+	}
 
-		parent_rect = DisplayServer::get_singleton()->screen_get_usable_rect(DisplayServer::get_singleton()->window_get_current_screen(w->get_window_id()));
+	// The DisplayServer doesn't allow resizing so there is no parent that we could resize into.
+	if (window_id != DisplayServer::INVALID_WINDOW_ID && DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_SELF_FITTING_WINDOWS)) {
+		return Rect2i(position, size);
 	}
-	return parent_rect;
+
+	// Internal popups should not overflow from the parent window.
```

Comment:
There's no reason for this limitation, popups can and do overflow parent all the time. If they could not, there would be zero point of supporting native window popups at all.